public/javascripts/BoardEditor.js

```diff
@@ -1,74 +1,116 @@
 let mouseX = 0;
 let mouseY = 0;
 let mouseDown = false;
 let lastTimestamp;
-let board = {};
+let board = {}; // list of drawingObjects
 let penSize = 3;
 let color = 'black';
-let left;
-
+let mouseLeft; // has the mouse left the board
+let newPoints = []; // the points that have been drawn that need to be emitted to other clients (for pen)
 const socket = io();
 const code = document.getElementById("code").value;
 
 socket.emit('join', { code: code });
-socket.on('joindata', function (data) {
+
+//  joinData
+//  Receive server broadcast to fill in the board on join
+//      data: 
+//      {
+//          nextId: current nextId for the board
+//          board: the board data for this board
+//      }
+socket.on('joinData', function (data) {
+    // Get the nextId for the board 
     nextId = data.nextId;
-    board = data.board;
+    // Get the board
+    sentBoard = data.board;
     let newBoard = {};
-    for (key in board) {
-        if (board[key].type == "Pen") {
-            newBoard[key] = new Pen(key, board[key].shapeData, { x: 0, y: 0 }, { x: 0, y: 0 }, board[key].size, board[key].color);
+    // Get the board objects
+    for (id in sentBoard) {
+        if (sentBoard[id].type == "Pen") {
+            newBoard[id] = new Pen(id, { x: 0, y: 0 }, { x: 0, y: 0 }, sentBoard[id].data.size, sentBoard[id].data.color);
+            newBoard[id].setPath(sentBoard[id].data.path);
         }
     }
     board = newBoard;
+    // Draw the objects
     compileBoard();
 });
-socket.emit('change', { code: code });
-socket.on('change', function (data) {
-    //console.log(data);
-});
-socket.on('add', function (data) {
+
+//  drawPen
+//  Receive server broadcast for drawing with the pen
+//      data: 
+//      {
+//          code: the code for the board being drawn on
+//          id: the id of the pen object
+//          size: the stroke-width of the pen object
+//          color: the color of the pen object
+//          newPoints: the points being added to the Pen object's path
+//      }
+socket.on('drawPen', function (data) {
+    // Make sure that the object being changed isn't being changed by this user rn
     if (nextId != data.id) {
-        if (data.type == "Pen") {
-            board[data.id] = new Pen(data.id, data.shapeData, { x: 0, y: 0 }, { x: 0, y: 0 }, data.size, data.color);
+        // If this board already has this Pen object, then update it accordingly
+        if (board[data.id]) {
+            board[data.id].updatePathData(data.newPoints);
+        }
+        // Otherwise add this Pen object to the board
+        else {
+            board[data.id] = new Pen(data.id, { x: 0, y: 0 }, { x: 0, y: 0 }, data.size, data.color);
+            board[data.id].updatePathData(data.newPoints);
         }
     }
     compileBoard();
 });
 
-socket.on('clearboard', function () {
+// clearBoard
+// Receive server broadcast to clear the board
+socket.on('clearBoard', function () {
     clearBoard();
 })
 
+//  newId
```

Comment:
I don't really care, but for future reference I think you should use /** */ comments for blocks like this

Reply:
at least i add comments 😈